contrib/docker/ckan-entrypoint.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+set -eu
+
+# URL for the primary database, in the format expected by sqlalchemy (required
+# unless linked to a container called 'db')
+: ${DATABASE_URL:=}
+# URL for solr (required unless linked to a container called 'solr')
+: ${SOLR_URL:=}
+# URL for redis (required unless linked to a container called 'redis')
+: ${REDIS_URL:=}
+
+CONFIG="${CKAN_CONFIG}/ckan.ini"
+
+abort () {
+  echo "$@" >&2
+  exit 1
+}
+
+write_config () {
+  ckan-paster make-config ckan "$CONFIG"
+
+  ckan-paster --plugin=ckan config-tool "$CONFIG" -e \
+      "sqlalchemy.url = ${DATABASE_URL}" \
+      "solr_url = ${SOLR_URL}" \
+      "ckan.redis.url = ${REDIS_URL}" \
+      "ckan.storage_path = ${CKAN_DATA}" \
+      "ckan.site_url = ${SITE_URL}"
```

Comment:
These are environment variables recognised by CKAN and used to override INI settings. See https://github.com/ckan/ckan/blob/master/ckan/config/environment.py#L117

If you use these environment variable names you don't need to generate an ini file with `paster config-tool` at all.